### PR TITLE
Add zlib 1.3

### PR DESCRIPTION
**Pull request by Vertexwahn**
Add zlib 1.3

### CODE REVIEW

File: modules/zlib/1.3/patches/add_build_file.patch [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
++        "trees.c",
++        "trees.h",
++        "uncompr.c",
++        "zconf.h",
++        "zutil.c",
++        "zutil.h",
++    ],
++    hdrs = ["zlib.h"],
++    copts = select({
++        "@bazel_tools//src/conditions:windows": [],
```
Comment (fmeum) [inline modules/zlib/1.3/patches/add_build_file.patch:39]: These conditions are being deprecated, could you use `@platforms//os:windows` instead?

**Review by fmeum — APPROVED** [review]

File: modules/zlib/1.3/patches/add_build_file.patch (hunk 1 above)
Comment (Vertexwahn) [reply modules/zlib/1.3/patches/add_build_file.patch]: I copied now the zlib.BUILD file from https://github.com/protocolbuffers/protobuf/blob/master/third_party/zlib.BUILD which has this issue fixed.

**Comment by fmeum** [comment]
@meteorcloudy Could you also take a look as this is a Bazel dep?

**Review by meteorcloudy — APPROVED** [review]
Building Bazel works with this version, and the BUILD file seems to be a better version.

**Pull request merged**